### ISSUE TITLE
Native Android variant freezes app when using either minimunDate, maximumDate and scroll over invalid dates

### ISSUE THREAD

**Issue by har0ld-scl** [henninghall/react-native-date-picker#295]
**Describe the bug**
Unmounting the date picker configured with `nativeAndroid` variant plus either `minimumDate`, `maximumDate` or both, while the wheel is still spinning will cause the app stop responding.

**Expected behavior**
The app shouldn't freeze when the component is unmounted in any conditions.

**To Reproduce**
```javascript
import { Component } from "react";
import DatePicker from "react-native-date-picker";
import { add as dateAdd, sub as dateSubtract } from "date-fns";

export default class App extends Component {
  state = {
    selectedDate: new Date(),
    pickerDate: new Date(),
    showModal: false,
  };

  render = () => (
    <>
      <View>
        <Button title={"Set Date"} onPress={this._showModal} />
        <Text>{this.state.selectedDate}</Text>
      </View>
      <Modal visible={this.state.showModal}>
        <DatePicker
          date={this.state.pickerDate}
          mode={"date"}
          androidVariant={"nativeAndroid"}
          minimumDate={dateSubtract(new Date(), { days: 1 })}
          maximumDate={dateAdd(new Date(), { days: 1 })}
          onDateChange={this._handleDateChange}
        />
        <View>
          <Button title={"Accept"} onPress={this._handleAccept} />
          <Button title={"Cancel"} onPress={this._hideModal} />
        </View>
      </Modal>
    </>
  );

  _showModal = () => {
    const { selectedDate } = this.state;
    this.setState({ showModal: true, pickerDate: selectedDate });
  };

  _hideModal = () => {
    this.setState({ showModal: false });
  };

  _handleDateChange = (date) => {
    this.setState({ pickerDate: date });
  };

  _handleAccept = () => {
    const { pickerDate } = this.state;
    this.setState({ selectedDate: pickerDate, showModal: false });
  };
}
```

Also I've created the following repo to reproduce the problem:
https://github.com/har0ld-scl/date-picker-crash

Steps to reproduce:
1. Add a date picker inside a modal.
2. Configure the date picker with `nativeAndroid` variant, use either `minimumDate`, `maximumDate` or both.
3. Open the modal.
4. Scroll the wheel over invalid dates (depending on the dates you've used in step 2).
5. Close the modal (ideally when the wheel is still spinning).
6. It should freeze at this time but if not repeat from steps 3 to 5 (sometimes it take more than one shot to reproduce).

![react-native-date-picker-freeze](https://user-images.githubusercontent.com/61238804/108083358-bb2ecb00-7051-11eb-897e-eeb52ffcdbd6.gif)

**Smartphone:**
 - OS: `Android`
 - React Native version: `0.63.4`
 - react-native-date-picker version: `3.2.9`

**Additional considerations**
- Only happens with the `nativeAndroid` variant. I've tested `iosClone` and it works fine.
- When you don't use neither `minimumDate` or `maximumDate` stop happening.
- I've tested in several devices (emulators and physical).

I have a clue why and where the problem is located. I will submit a pull request with the proposal fix when my test gives positive results.

Thanks!

**Comment by henninghall** [comment]
@har0ld-scl  Thanks for that PR, the fix is released in `v3.2.10`

**Comment by har0ld-scl** [comment]
@henninghall  Awesome news! 

I’m happy to help.
